subprojects/packagefiles/sfml/meson.build

```diff
@@ -6,22 +6,20 @@ project(
     'cpp_std=c++17',
     'wrap_mode=forcefallback',
   ],
-  meson_version: '>=0.49.0',
+  meson_version: '>=0.59.0',
 )
 
 cpp = meson.get_compiler('cpp')
 
-openal_dep = dependency('openal')
-thread_dep = dependency('threads')
+graphics_opt = get_option('graphics')
+audio_opt = get_option('audio')
+system_opt = get_option('system')
+window_opt = get_option('window')
+network_opt = get_option('network')
+
 gl_dep = dependency('gl')
 winmm_dep = cpp.find_library('winmm', required: host_machine.system() == 'windows')
 w32_dep = cpp.find_library('ws2_32', required: host_machine.system() == 'windows')
-freetype_dep = dependency('freetype2')
-flac_dep = dependency('flac')
-ogg_dep = dependency('ogg')
-vorbis_dep = dependency('vorbis')
-vorbisenc_dep = dependency('vorbisenc')
-vorbisfile_dep = dependency('vorbisfile')
```

Comment:
Better to keep those checks here, and taking the project options into account.

For example, are the audio requirements hard dependencies?

Reply:
audio is the only library you can use apart from
in reality graphics needs system and windows (i guess that's my mistake )

Reply:
Ah !!! I get it you want me to leave them in the main meson but check the option to add them?

Reply:
It looks like the cmake build system helper for external dependency (`sfml_find_package`) is using `find_package(… REQUIRED)`, so they're all mandatory.

Therefore, I would move back those checks here, for example:

```diff
diff --git i/releases.json w/releases.json
index 33560837..93fee497 100644
--- i/releases.json
+++ w/releases.json
@@ -3022,7 +3022,6 @@
   },
   "sfml": {
     "dependency_names": [
-      "sfml",
       "sfml-all",
       "sfml-graphics",
       "sfml-system",
diff --git i/subprojects/packagefiles/sfml/meson.build w/subprojects/packagefiles/sfml/meson.build
index a03f65d2..ff8af273 100644
--- i/subprojects/packagefiles/sfml/meson.build
+++ w/subprojects/packagefiles/sfml/meson.build
@@ -16,6 +16,19 @@ audio_opt = get_option('audio')
 window_opt = get_option('window')
 network_opt = get_option('network')
 
+foreach audio_dep : [
+  'flac',
+  'ogg',
+  'openal',
+  'vorbis',
+  'vorbisenc',
+  'vorbisfile',
+  ]
+  _dep = dependency(audio_dep, required: audio_opt)
+  audio_opt = audio_opt.require(_dep.found())
+  set_variable(audio_dep + '_dep', _dep)
+endforeach
+
 gl_dep = dependency('gl')
 winmm_dep = cpp.find_library('winmm', required: host_machine.system() == 'windows')
 w32_dep = cpp.find_library('ws2_32', required: host_machine.system() == 'windows')
@@ -41,3 +54,15 @@ meson.override_dependency('sfml', sfml_all_dep)
 if get_option('examples')
   subdir('examples')
 endif
+
+summary(
+  {
+    'audio'   : get_option('audio').allowed(),
+    'graphics': get_option('graphics').allowed(),
+    'network' : get_option('network').allowed(),
+    'window'  : get_option('window').allowed(),
+  },
+  bool_yn: true,
+  section: 'Features',
+)
+
diff --git i/subprojects/packagefiles/sfml/src/SFML/Audio/meson.build w/subprojects/packagefiles/sfml/src/SFML/Audio/meson.build
index 34ef2bef..dc6c4807 100644
--- i/subprojects/packagefiles/sfml/src/SFML/Audio/meson.build
+++ w/subprojects/packagefiles/sfml/src/SFML/Audio/meson.build
@@ -22,13 +22,6 @@ audio_sources = files(
   'SoundStream.cpp',
 )
 
-openal_dep = dependency('openal')
-flac_dep = dependency('flac')
-vorbis_dep = dependency('vorbis')
-vorbisenc_dep = dependency('vorbisenc')
-vorbisfile_dep = dependency('vorbisfile')
-ogg_dep = dependency('ogg')
-
 audio_lib = library(
   'sfml-audio',
   audio_sources,
```

Reply:
And only export `sfm-all` if all sub-systems are available, I think:
```meson
if (
  audio_opt.allowed()
  and graphics_opt.allowed()
  and network_opt.allowed()
  and window_opt.allowed()
)
  sfml_all_dep = declare_dependency(
    include_directories: pub_inc,
    link_with: [host_machine.system() == 'windows' ? main_lib : []],
    dependencies: [
      sfml_audio_dep,
      sfml_graphics_dep,
      sfml_network_dep,
      sfml_window_dep,
      sfml_system_dep,
    ],
  )
  meson.override_dependency('sfml-all', sfml_all_dep)
  meson.override_dependency('sfml', sfml_all_dep)
endif

```